pkg/asset/agent/agentconfig/agent_config_test.go

```diff
@@ -12,9 +12,76 @@ import (
 	"github.com/openshift/installer/pkg/types/baremetal"
 	"github.com/pkg/errors"
 	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func TestAgentConfig_Generate(t *testing.T) {
+
+	cases := []struct {
+		name           string
+		expectedError  string
+		expectedConfig *agent.Config
+	}{
+		{
+			name: "generate-basic-template",
+			expectedConfig: &agent.Config{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "AgentConfig",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "example-agent-config",
+					Namespace: "cluster0",
+				},
+				Spec: agent.Spec{
+					RendezvousIP: "your-node0-ip",
+					Hosts: []agent.Host{
+						{
+							Hostname: "change-to-hostname",
+							Role:     "master",
+							RootDeviceHints: baremetal.RootDeviceHints{
+								DeviceName: "/dev/sda",
+							},
+							Interfaces: []*aiv1beta1.Interface{
+								{
+									Name:       "your-network-interface-name",
+									MacAddress: "00:00:00:00:00",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			parents := asset.Parents{}
+			asset := &Asset{}
```

Comment:
Shouldn't asset be above parents here?

Reply:
The line above is referring to the package `assets`, not this object, which is shadowing the name. (It might pay to rename this to agentConfig though.)

Reply:
Thanks was a just a quick look while I was making coffee and it looked off to me.